### PR TITLE
Use uv project interface in template

### PR DESCRIPTION
**Pull request by blueraft**
- Blog [post](https://astral.sh/blog/uv-unified-python-packaging#projects) announcing `uv` project interface. 
- [Guide](https://docs.astral.sh/uv/guides/projects/) on how to manage projects.

It's still compatible with standard pip if a user doesn't want to use `uv`.

### CODE REVIEW

File: {{cookiecutter.plugin_name}}/pyproject.toml [hunk 1]
```diff
@@ -111,6 +138,11 @@ skip-magic-trailing-comma = false
 # Like Black, automatically detect the appropriate line ending.
 line-ending = "auto"
 
+[tool.uv]
```
Comment (JosePizarro3) [inline {{cookiecutter.plugin_name}}/pyproject.toml:141]: This is not always working in MacOS (I think depending on the version) and Windows, and in Ubuntu Bernadette was also having the same issue. I saw all of this during the hackathon last week

File: {{cookiecutter.plugin_name}}/README.md [hunk 2]
```diff
@@ -157,15 +136,21 @@ pip install dist/{{cookiecutter.plugin_name}}-{{cookiecutter.version}}
 Read more about python packages, `pyproject.toml`, and how to upload packages to PyPI
 on the [PyPI documentation](https://packaging.python.org/en/latest/tutorials/packaging-projects/).
 
-
 ### Template update
 
-We use cruft to update the project based on template changes. A `cruft-update.yml` is included in Github workflows to automatically check for updates and create pull requests to apply updates. Follow the [instructions](https://github.blog/changelog/2022-05-03-github-actions-prevent-github-actions-from-creating-and-approving-pull-requests/) on how to enable Github Actions to create pull requests. 
+We use cruft to update the project based on template changes. A `cruft-update.yml` is included in Github workflows to automatically check for updates and create pull requests to apply updates. Follow the [instructions](https://github.blog/changelog/2022-05-03-github-actions-prevent-github-actions-from-creating-and-approving-pull-requests/) on how to enable Github Actions to create pull requests.
```
Comment (JosePizarro3) [inline {{cookiecutter.plugin_name}}/README.md:141]: This cruft update is never working (as the changes are a bit too hard to track), so I doubt anyone is ever going to accept these changes, and the email notifications are a bit annoying.

File: {{cookiecutter.plugin_name}}/README.md [hunk 3]
```diff
@@ -4,87 +4,70 @@
...
 cd {{cookiecutter.plugin_name}}
-python3.11 -m venv .pyenv
-. .pyenv/bin/activate
 ```
 
-Make sure to have `pip` upgraded:
-```sh
-pip install --upgrade pip
-```
+We highly recommend using `uv` to manage the project. Consider installing `uv` by following the official installation [guide](https://docs.astral.sh/uv/getting-started/installation/#installing-uv).
```
Comment (JosePizarro3) [inline {{cookiecutter.plugin_name}}/README.md:16]: Please, do not send people people to read some other documentation in another package management. Can you add a better explanation + step-by-step as it was before? Think that people is not going to open that uv guide (not even me), so is `uv` replacing the virtual environment? Can you summarize how uv manages projects in 1 or 2 lines?

File: {{cookiecutter.plugin_name}}/README.md [hunk 4]
```diff
@@ -4,87 +4,70 @@
...
 
-We recommend installing `uv` for fast pip installation of the packages:
-```sh
-pip install uv
-```
+Read the short [guide](https://docs.astral.sh/uv/guides/projects/) on how to manage projects.
```
Comment (JosePizarro3) [inline {{cookiecutter.plugin_name}}/README.md:18]: See previous comment.

File: {{cookiecutter.plugin_name}}/README.md [hunk 5]
```diff
@@ -4,87 +4,70 @@
...
+Install all the dependencies:
 
-**Note!**
-Until we have an official pypi NOMAD release with the plugins functionality make
-sure to include NOMAD's internal package registry (via `--index-url` in the above command).
-
-The plugin is still under development. If you would like to contribute, install the package in editable mode (with the added `-e` flag):
 ```sh
-uv pip install -e '.[dev]' --index-url https://gitlab.mpcdf.mpg.de/api/v4/projects/2187/packages/pypi/simple
+uv sync --extra-all
```
Comment (JosePizarro3) [inline {{cookiecutter.plugin_name}}/README.md:23]: Ok, this is new and I wrote a comment before regarding the tool.uv part with the index-url. So maybe you can ask Bernadette if this works for her? And maybe others in the team?

**Review by JosePizarro3 — APPROVED** [review]
So now we can use `uv` directly without creating a virtual environment? Do we need to do some previous step to do this (like for generating the uv.lock file or something)?

**Comment by lauri-codes** [comment]
This should probably be discussed in the plugin meeting so that people can voice their opinion.

File: {{cookiecutter.plugin_name}}/pyproject.toml (hunk 1 above)
Comment (blueraft) [reply {{cookiecutter.plugin_name}}/pyproject.toml:141]: Are you sure they were not using just `pip` here? This is just the equivalent to passing the argument in the CLI.

File: {{cookiecutter.plugin_name}}/README.md (hunk 2 above)
Comment (blueraft) [reply {{cookiecutter.plugin_name}}/README.md:141]: I agree, maybe we can just keep the local command and remove the CI step.

**Comment by blueraft** [comment]
> So now we can use uv directly without creating a virtual environment? Do we need to do some previous step to do this 

Yes, you won't have to manually manage the environment. `uv sync`/ `uv run` for the first time will also create the lock file for you. All invocations of `uv run` will automatically use the correct venv and your venv will be up to date with all of the changes in the `pyproject.toml` file. 

> This should probably be discussed in the plugin meeting so that people can voice their opinion.

Makes sense, I will create a short intro for this and how to develop multiple plugins locally. In the mean time maybe Chema and others can try it out and see if it's helpful.
